src/renderer/components/outputs/LargeImageOutput.tsx

```diff
@@ -30,120 +35,251 @@ const pickImage = (previews: PreviewImage[], realSize: number) => {
     return found ?? previews[previews.length - 1];
 };
 
-export const LargeImageOutput = memo(({ output, useOutputData, animated }: OutputProps) => {
-    const { t } = useTranslation();
+export const LargeImageOutput = memo(
+    ({ output, useOutputData, animated, size, setSize }: OutputProps) => {
+        const { t } = useTranslation();
 
-    const dpr = useDevicePixelRatio();
-    const zoom = useContextSelector(GlobalVolatileContext, (c) => c.zoom);
-    const realSize = IMAGE_PREVIEW_SIZE * zoom * dpr;
+        const dpr = useDevicePixelRatio();
+        const zoom = useContextSelector(GlobalVolatileContext, (c) => c.zoom);
+        const realSize = (size?.width ?? IMAGE_PREVIEW_SIZE) * zoom * dpr;
 
-    const { last, stale } = useOutputData<LargeImageBroadcastData>(output.id);
+        const { last, stale } = useOutputData<LargeImageBroadcastData>(output.id);
 
-    const imgBgColor = 'var(--node-image-preview-bg)';
-    const fontColor = 'var(--node-image-preview-color)';
+        const imgBgColor = 'var(--node-image-preview-bg)';
+        const fontColor = 'var(--node-image-preview-color)';
 
-    const pickedImage = last ? pickImage(last.previews, realSize) : null;
+        const previewImage = last ? pickImage(last.previews, realSize) : null;
 
-    return (
-        <Center
-            h="full"
-            minH="2rem"
-            overflow="hidden"
-            verticalAlign="middle"
-            w="full"
-        >
+        const { useSnapToGrid } = useContext(SettingsContext);
+        const [isSnapToGrid, , snapToGridAmount] = useSnapToGrid;
+
+        const [resizeRef, setResizeRef] = useState<Resizable | null>(null);
+
+        const [maxSize, setMaxSize] = useState<Size>({
+            width: IMAGE_PREVIEW_SIZE,
+            height: IMAGE_PREVIEW_SIZE,
+        });
+
+        useEffect(() => {
+            if (last?.previews) {
+                const biggestImage = last.previews[last.previews.length - 1];
+                const img = new window.Image();
+                img.src = biggestImage.url;
+                img.onload = () => {
+                    setMaxSize({
+                        width: img.width,
+                        height: img.height,
+                    });
```

Comment:
Q: Does this make sense? Why should users be allowed to make a comically large View Image node?

Practically speaking, it's also not very user-friendly. Image setting up a chain to your liking, including a large View Image node, and then chainner just auto-shrinks the node when you run the chain with a small image. After all, people that work on small pixel art sprites would probably like to see their sprite in big.

Reply:
So you're suggesting we scale the image to the size (maintaining aspect ratio) no matter what?

Reply:
And keeping the max size something like 1024x instead?

Reply:
> So you're suggesting we scale the image to the size (maintaining aspect ratio) no matter what?

Yes, but I just realized that we didn't do that before. We just centered small images before. Just centering small images would be fine too.

But I was mainly saying that resizing the LargeImageOutput based on the preview image size is bad.

Reply:
Gotcha. Well, I could certainly make it so it's always something like a 216-1024 range and then we just center anything smaller

Reply:
Sounds good.